src/Core/PackageKitBackend.vala

```diff
@@ -769,6 +795,59 @@ public class AppCenterCore.PackageKitBackend : Backend, Object {
         return (PackageDetails)job.result.get_object ();
     }
 
+    private void ProgressCallback (Pk.Progress progress, Pk.ProgressType type) {
```

Comment:
At first I didn't understand why this method was called in the callbacks of calls to PackageKit, we should perhaps rename it to something more descriptive like `update_progress_status ()` since it updates global state of the backend.